x-pack/plugins/infra/server/lib/create_custom_metrics_aggregations.ts

```diff
@@ -0,0 +1,76 @@
+/*
```

Comment:
I'm wondering if this file should be in `metrics_explorer/lib` instead. What do you think?

Reply:
Oh... I struggled to decide what's the most appropriate place for this file. It's being used in both the Metrics Explorer and Metric Threshold rule. The only reason the typing on the `customMetrics` variable is `MetricsExplorerCustomMetric[]` is because that's what I started with. I wonder if we shouldn't type `customMetrics` with `MetricsExplorerCustomMetric[] | MetricExpressionCustomMetric[]` and then detect the difference internally so it's more ubiquitous. The other thought I had is "maybe we should just change the definition for a `customMetric` to be the same for both so they are interchangeable and we don't need to convert using `convertToMetricExplorerCustomMetric`. 

BTW the only difference between the 2 formats is that the aggregation property name is `aggregation` in Metrics Explorer and `aggType` in Metric Threshold Rule. I was trying to keep the `customMetric` structure as consistent as the rest of the names within their respective context.

Reply:
OK... I decided to leave it where it is but change the definition for `customMetrics` to `MetricsExplorerCustomMetric[] | MetricExpressionCustomMetric[]` and handle the conversion inside `createCustomMetricsAggregations`. This way it works for both types and the types can remain consistent in their respective contexts (`aggregation` vs `aggType`).